source/uwp/Renderer/lib/XamlBuilder.cpp

```diff
@@ -389,6 +393,13 @@ namespace AdaptiveNamespace
                 ComPtr<IUIElement> separator;
                 XamlHelpers::AddSeparatorIfNeeded(iElement, element, hostConfig.Get(), renderContext, parentPanel, &separator);
 
+                // If the renderedElement was an input, render the label and error message
```

Comment:
>// If the renderedElement was an input, render the label and error message [](start = 16, length = 74)

Why do we do this here rather than calling this function from inside the input renderers? What happens in the custom rendering case? If i were making a custom input renderer i would expect that rendering the label would be part of my job...

Reply:
we made the decision that label and error messages would be respected by us, at some point in the future we're planning on providing a way to override the rendering behaviour for labels and error messages. It's similar to how we manage the spacing and input height for elements instead of waiting for the card authors to do it by themselves

---
In reply to: [445152335](https://github.com/microsoft/AdaptiveCards/pull/4088#discussion_r445152335) [](ancestors = 445152335)

Reply:
Is this mentioned in the spec? If not, please make sure to document this behavior there (it seems non-intuitive to me)

---
In reply to: [445220327](https://github.com/microsoft/AdaptiveCards/pull/4088#discussion_r445220327) [](ancestors = 445220327,445152335)